packages/test-base/src/nativeDarwin/kotlin/io/realm/kotlin/test/platform/PlatformUtils.kt

```diff
@@ -34,10 +34,12 @@ import kotlin.time.Duration
 
 actual object PlatformUtils {
     actual fun createTempDir(prefix: String, readOnly: Boolean): String {
-        // X is a special char which will be replace by mkdtemp template
-        val mask = prefix.replace('X', 'Z', ignoreCase = true)
-        val path = "${platform.Foundation.NSTemporaryDirectory()}$mask"
-        platform.posix.mkdtemp(path.cstr)
+        // Currently we cannot template using platform.posix.mkdtemp
+        // the return value is not of use.
+        val suffix = "-${Utils.createRandomString(6)}"
+        val path = "${platform.Foundation.NSTemporaryDirectory()}$prefix$suffix"
+        platform.posix.mkdir(path, 448.toUShort())
```

Comment:
maybe using the const like in https://github.com/realm/realm-kotlin/pull/1550/files#diff-0e15d894eafc8be3bb7b226a9802e7a2ec938f153158929d9377c6b5635316e1R44 is more readable than `448`